octue/runner.py

```diff
@@ -194,16 +194,35 @@ class AppFrom:
 
     def __init__(self, app_path="."):
         self.app_path = os.path.abspath(os.path.normpath(app_path))
+        module_logger.debug(f"Initialising AppFrom context at app_path {self.app_path}")
         self.app_module = None
 
     def __enter__(self):
+        # Warn on an app present on the system path
+        if "app" in sys.modules.keys():
+            module_logger.warning(
+                "Module 'app' already on system path. Using 'AppFrom' context will yield unexpected results. Avoid using 'app' as a python module, except for your main entrypoint"
+            )
+
+        # Insert the present directory first on the system path
         sys.path.insert(0, self.app_path)
+
+        # Import the app from the present directory
         self.app_module = importlib.import_module("app")
+
+        # Immediately clean up the entry to the system path (don't use "remove" because if the user has it in their
+        # path, this'll be an unexpected side effect, and don't do it in cleanup in case the called code inserts a path)
+        sys.path.pop(0)
+        module_logger.debug(
+            f"Imported app at app_path and cleaned up temporary modification to sys.path {self.app_path}"
+        )
+
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
-        if self.app_path in sys.path:
-            sys.path.remove(self.app_path)
```

Comment:
So does `app` need to be cleared from `sys.path` as well as from `sys.modules`? Or just `sys.modules?

Reply:
The present directory does need to be removed form sys.path, but look a couple of lines up - I `pop` it from the path immediately after it's been used for import.